src/scene/shader-lib/chunks/lit/frag/shadowPCSS.js

```diff
@@ -0,0 +1,185 @@
+export default /* glsl */`
+
+
+#define PCSS_SAMPLE_COUNT 16
+
+vec2 VogelDisk(float sampleIndex, float count, float phi)
+{
+    const float GoldenAngle = 2.4;
+    float r = sqrt(sampleIndex + 0.5) / sqrt(count);
+
+    float theta = sampleIndex * GoldenAngle + phi;
+
+    float sine, cosine;
+    sine = sin(theta);
+    cosine = cos(theta);
+
+    return vec2(r * cosine, r * sine);
+}
+
+vec3 VogelSphere(float sampleIndex, float count, float phi) {
+    float weight = sampleIndex / count;
+    float radius = sqrt(1.0 - weight * weight);
+    const float GoldenAngle = 2.4;
+
+    float theta = GoldenAngle * sampleIndex + phi;
+
+    return vec3(cos(theta) * radius, weight, sin(theta) * radius);
+}
+
+float GradientNoise(vec2 screenPos) {
+    vec3 magic = vec3(0.06711056, 0.00583715, 52.9829189);
+    return fract(magic.z * fract(dot(screenPos, magic.xy)));
+}
+
+#ifndef UNPACKFLOAT
+#define UNPACKFLOAT
+float unpackFloat(vec4 rgbaDepth) {
+    const vec4 bitShift = vec4(1.0 / (256.0 * 256.0 * 256.0), 1.0 / (256.0 * 256.0), 1.0 / 256.0, 1.0);
+    return dot(rgbaDepth, bitShift);
+}
+#endif
+
+float viewSpaceDepth(float depth, mat4 invProjection) {
+    float z = depth * 2.0 - 1.0;
+    vec4 clipSpace = vec4(0.0, 0.0, z, 1.0);
+    vec4 viewSpace = invProjection * clipSpace;
+    return viewSpace.z;
+}
+
+
+float PCSSBlockerDistance(TEXTURE_ACCEPT(shadowMap), vec2 sampleCoords[PCSS_SAMPLE_COUNT], vec2 shadowCoords, float searchSize, float z) {
+
+    float blockers = 0.0;
+    float averageBlocker = 0.0;
+    for (int i = 0; i < PCSS_SAMPLE_COUNT; i++) {
+        vec2 offset = sampleCoords[i] * searchSize;
+        vec2 sampleUV = shadowCoords + offset;
+
+    #ifdef GL2
+        float blocker = texture2D(shadowMap, sampleUV).r;
+    #else // GL1
+        float blocker = unpackFloat(texture2D(shadowMap, sampleUV));
+    #endif        
+        float isBlocking = step(blocker, z);
+        blockers += isBlocking;
+        averageBlocker += blocker * isBlocking;
+    }
+
+    if (blockers > 0.0)
+        return averageBlocker /= blockers;
+    return -1.0;
+}
+
+float PCSS(TEXTURE_ACCEPT(shadowMap), vec3 shadowCoords, vec4 cameraParams, float oneOverShadowMapSize, float lightSize) {
+    float receiverDepth = linearizeDepth(shadowCoords.z, cameraParams);
+#ifndef GL2
+    // If using packed depth on GL1, we need to normalize to get the correct receiver depth
+    receiverDepth *= 1.0 / (cameraParams.y - cameraParams.z);
+#endif
+
+    vec2 samplePoints[PCSS_SAMPLE_COUNT];
+    float noise = GradientNoise( gl_FragCoord.xy ) * 2.0 * PI;
+    for (int i = 0; i < PCSS_SAMPLE_COUNT; i++) {
+        samplePoints[i] = VogelDisk(float(i), float(PCSS_SAMPLE_COUNT), noise);
+    }
+
+    // Calculate the ratio of FOV between 45.0 degrees (tan(45) == 1) and the FOV of the camera    
+    float fovRatioAtDepth = receiverDepth / (cameraParams.x * receiverDepth);
```

Comment:
don't `receiverDepth`'s cancel each other here? So the result is `1/cameraParams.x`

Reply:
True, simplified it.